### PR TITLE
Feature/demos improvement

### PR DESCRIPTION
**Pull request by anton-karlovskiy**
@addyosmani 

React Twitter projects are updated with new mobile view check logic.
React Youtube project is updated with debugging option.
The demos using react-toggle package are updated regarding styling.

React Twitter projects and React Youtube project have Glitch versions in README.

Hooks directory is removed as we have `react-adaptive-hooks` package.

### CODE REVIEW

**Comment by anton-karlovskiy** [comment]
@addyosmani 

Static data for Youtube demo is up-to-date.
cna-memory-considerate-animation is updated with Arial font family.

**Comment by anton-karlovskiy** [comment]
@addyosmani 

Query param debugging option is implemented on Youtube demo.
https://adaptive-loading.web.app/react-youtube-adaptive-loading/watch?v=OTCOoVApM_U&mode=lite

Available mode values are `lite` and `full`.

**Comment by addyosmani** [comment]
Thank you! At some point it would be great to chat about the JavaScript bundles for the YouTube watch page, but even just looking at the difference between lite vs. full is kind of amazing: https://www.webpagetest.org/video/compare.php?tests=191126_7S_dee60a379e7d94249b3ba7f49ba8582c,191126_J3_b9181c15eb2909081f1548d87274c3b8

Appreciate you adding the flags!

**Comment by anton-karlovskiy** [comment]
> At some point it would be great to chat about the JavaScript bundles for the YouTube watch page.

I think I have not got it fully. Could you please be more specific? :)

File: react-twitter-save-data-loading(hook)/src/App.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ import Tweet from './components/Twitter/Tweet';
 import Navbar from './components/Navbar/Navbar';
 import tweets from './data/tweets';
 import { IMAGE_TYPE } from './config';
-import { checkMobile } from './utils/helpers';
+import { checkIsSmallViewport } from './utils/helpers';
```
Comment (addyosmani) [inline react-twitter-save-data-loading(hook)/src/App.js]: Noting feedback from chat: the existing `checkMobile` test was relying on a combination of UA strings in order to determine if we were on a mobile or desktop, however, as we know such strings are fragile and cannot be relied on when browsers often fake UA strings. For now we've settled on a solution that relies on matchMedia for checks.

File: react-youtube-adaptive-loading/src/components/Debugging/CheckboxWithLabel/CheckboxWithLabel.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/*
```
Comment (addyosmani) [inline react-youtube-adaptive-loading/src/components/Debugging/CheckboxWithLabel/CheckboxWithLabel.js]: The purpose of this debugging option was to make it trivial for developers to toggle the lite or full mode without needing to trip into one of the hardware/network conditions required for experiencing one or the other. We are also adding a query-parameter for making this switch to enable simple benchmarking use-cases.

File: react-youtube-adaptive-loading/src/App.js [hunk 3]
```diff
@@ -41,7 +41,9 @@ const App = ({ youtubeLibraryLoaded, location }) => {
   } = useLiteModeDebugging();
 
   useEffect(() => {
-    loadYoutubeApi();
+    if (!YOUTUBE_API_DEV_MODE) {
```
Comment (addyosmani) [inline react-youtube-adaptive-loading/src/App.js]: For later: due to YouTube API restrictions we have landed on a model where we can optionally use static data for testing or live data via an API key if we prefer.

**Comment by addyosmani** [comment]
> > At some point it would be great to chat about the JavaScript bundles for the YouTube watch page.
> 
> I think I have not got it fully. Could you please be more specific? :)

Certainly. If you take a look at the above WebPageTest timeline, even the lite version takes many seconds before we can see any pixels painted. While we are only using 152KB of JS to render the page and it's broken up to an extent, we might want to continue looking at opportunities to improve it. Nothing major for now however. I should re-test on 4G :)

**Pull request merged**